packages/next/src/middleware/request.ts

```diff
@@ -1,9 +1,22 @@
 import type { NextURL } from 'next/dist/server/web/next-url'
 import { NextResponse } from 'next/server'
-import type { NextRequest } from 'next/server'
+import type { NextRequest as InternalNextRequest } from 'next/server'
 
 import { MiddlewareResponse } from './response'
 
+export type NextRequest = InternalNextRequest & {
+  get geo():
+    | {
+        city?: string
+        country?: string
+        region?: string
+        latitude?: string
+        longitude?: string
+        timezone?: string
+      }
+    | undefined
+}
+
```

Comment:
[dust] Rather than redefining the built-in geo type, how about extending it. Less verbose and easier to maintain. e.g. 

```suggestion
export type NextRequest = InternalNextRequest & {
  geo: InternalNextRequest['geo'] & {
    timezone?: string
  }
}

```

Reply:
For some reason I thought I had to reimplement the whole thing as it was an actual getter. Didn't realize you could jus tag that on like that like a merge declaration. I'll go ahead and make the change.